### DISCUSSION TITLE
how to use images properly

### DISCUSSION THREAD

**Discussion by kkerti** [alex8088/electron-vite#89] [Q&A] [answered]
Hi,
I can't get my head around using images in `<img src="/assets/imgs/.."/>` properly. It works in dev, but the built electron app has no images. During `electron-vite build` the images aren't copied to dist. I think I'm missing some config, but after reading the docs I'm still clueless.

Here is my folder structure and the current renderer config.

Any help would be appreciated. Thanks.

![image](https://user-images.githubusercontent.com/47832952/213524649-e87b6884-2147-42cf-b6a3-54b36e727413.png)
```
renderer: {
    plugins: [
      svelte({
        preprocess: [
          preprocess({
            postcss: true,
          })
        ]
      }), 
      monacoEditorPlugin([])
    ],
    build: {
      rollupOptions: {
        input: {
          index: resolve(__dirname, 'src/renderer/index.html'),
        }
      },
      outDir: '../../dist/renderer',
    },
    resolve: {
      alias: {
        '$lib': path.resolve('src/renderer/lib'),
      },
    },
    target: 'chrome104'
  }   
```

**Comment by alex8088** [comment, marked as answer]
use assets via importing, for example:
```
import icon from './assets/icons.svg'
<img src="{icon}"/>
```

**Comment by kkerti** [comment]
Thanks alex. Does a dynamic import work in this case as well?

**Reply by alex8088** [reply]
> Thanks alex. Does a dynamic import work in this case as well?

yes

**Reply by kkerti** [reply]
thank you alex!

**Reply by kkerti** [reply]
@alex8088 
Using the following function, development works, but prod build does not get the images.

```
function getImgUrl(img){
    const imgPath = new URL(`/assets/imgs/sm_${img}.jpg`).href
    console.log('img', img, imgPath)
    return imgPath
  }
  ```
  Any ideas?
  

![image](https://user-images.githubusercontent.com/47832952/213683788-8f91fe3d-b16a-41ab-b686-bdbbe54d4e21.png)

**Reply by alex8088** [reply]
> @alex8088 Using the following function, development works, but prod build does not get the images.
> 
> ```
> function getImgUrl(img){
>     const imgPath = new URL(`/assets/imgs/sm_${img}.jpg`).href
>     console.log('img', img, imgPath)
>     return imgPath
>   }
> ```
> 
> Any ideas?
> 
> ![image](https://user-images.githubusercontent.com/47832952/213683788-8f91fe3d-b16a-41ab-b686-bdbbe54d4e21.png)

pass `import.meta.url` to URL as a second arg. 

You can learn more from vite's static assets handling section.
